### PR TITLE
feat/refactor: add `/query_with_image` route

### PR DESCRIPTION
**Pull request by khayliang**
This PR introduces a major new route `/query_with_image` and a minor route `/image_to_text`

- `/query_with_image` allows `Examplify` to accept images as a new data structure.

- `/image_to_text` is a route for debugging the OCR, similar to `/generate`

Heavy refactoring was also done to the codebase to allow future extensions for querying with other data structures.

Major fixes to be compatible with the new `zephyr-7b` model was also added.

### CODE REVIEW

File: server/features/extraction/extract_text.py [hunk 1]
```diff
@@ -36,6 +38,30 @@ def extract_text(file_name: str, file_type: str, file: bytes) -> Document:
...
+    """
+    with PyTessBaseAPI() as ocr:
+        with Image.open(file) as img:
+            sections = [ocr.image_to_text(img)]
+
+    return Document(
+        id=str(uuid4()),
+        sections=sections,
+        semantic_identifier=file_name
+    )
```
Comment (winstxnhdw) [inline server/features/extraction/extract_text.py:64]: I think we should abstract this out into it's own file. We will probably have a `extract_document_from_image` and a `extract_text_from_image`. The former is for creating an embedding and the latter will be for querying with an image.

Comment (winstxnhdw) [reply server/features/extraction/extract_text.py:64]: I'll probably split it into `extract_from_image.py` and then rename the current one to `extract_from_pdf.py`.

Comment (khayliang) [reply server/features/extraction/extract_text.py:64]: ill work on that

**Comment by khayliang** [comment]
@winstxnhdw I moved the code for OCR out to a separate file. Also added package requirements into the dockerfile.

**Comment by khayliang** [comment]
Created endpoint for querying with image. Left with duplicate code:

```
************* Module server.lifespans.download_embeddings
server/lifespans/download_embeddings.py:1:0: R0801: Similar lines in 2 files
==server.api.v1.query:[32:47]
==server.api.v1.query_from_image:[41:56]
    )

    search_response = await redis.ft(Config.redis_index_name).search(
        redis_query,
        redis_query_parameters  # type: ignore  (this is a bug in the redis-py library)
    )

    context = ' '.join(
        document['content'] for document
        in search_response.docs # type: ignore
    )

    message_history: list[Message] = await redis_get(redis, f'chat:{chat_id}', _ := [])
    message_history.append({
        'role': 'user', (duplicate-code)

------------------------------------------------------------------
```

**Comment by winstxnhdw** [comment]
So currently to remove the duplicate code, I have abstracted the query into a feature, but this is bad practice as it now contains database logic. Ideally, database logic is handled within a dependency or at the route-level. I'll have to refactor this in the future.

**Comment by winstxnhdw** [comment]
@khayliang Please allow workflows on your fork so we can run the status checks. Go here https://github.com/khayliang/Examplify/actions

**Comment by winstxnhdw** [comment]
Ok, I managed to abstract the database logic but it was pretty painful, and not very obvious.

**Comment by winstxnhdw** [comment]
LGTM. Vamos! 🎉

**Pull request merged**